util/conn/manager.go

```diff
@@ -132,14 +132,6 @@ func dial(d Dialer, network, address string, logger log.Logger) net.Conn {
 	return conn
 }
 
-func exponential(d time.Duration) time.Duration {
-	d *= 2
-	if d > time.Minute {
-		d = time.Minute
-	}
-	return d
-}
```

Comment:
I'm thinking a 1-line change can be enough, something like this, feel free to make corrections...

```go
func exponential(d time.Duration) time.Duration {
    d *= 2
    if d > time.Minute {
        d = time.Minute
    }
    return d * (rand.Float64() + 0.5)
}
```